samples/corda/corda-simple-application/clients/src/main/kotlin/com/cordaSimpleApplication/client/AssetTransferManager.kt

```diff
@@ -0,0 +1,763 @@
+            val partyName = rpc.proxy.startFlow(::GetOurIdentity).returnValue.get()
+            println("Name of the party owning the Corda node: $partyName")
+        } catch (e: Exception) {
+            println("Error: ${e.toString()}")
+        } finally {
+            rpc.close()
+        }
+    }
+}
+
+/**
+ * Command to reclaim a pledged asset after timeout.
+ */
+class ReclaimAssetCommand : CliktCommand(name="reclaim-pledged-asset", help = "Reclaims a pledged asset after timeout.") {
+    val config by requireObject<Map<String, String>>()
+    val pledgeId: String? by option("-pid", "--pledge-id", help="Pledge id for asset transfer pledge state")
+    val transferCategory: String? by option("-tc", "--transfer-category", help="transferCategory is input in the format: 'asset_type.remote_network_type'."
+        + " 'asset_type' can be either 'bond', 'token' or 'house-token'."
+        + " 'remote_network_type' can be either 'fabric', 'corda' or 'besu'.")
+    val importNetworkId: String? by option ("-inid", "--import-network-id", help="Import network id of pledged asset for asset transfer")
+    val exportRelayAddress: String? by option ("-era", "--export-relay-address", help="Asset export network relay address")
+    val param: String? by option("-p", "--param", help="Parameter AssetType:AssetId for non-fungible, AssetType:Quantity for fungible.")
+    val observer: String? by option("-o", "--observer", help="Party Name for Observer (e.g., 'O=PartyA,L=London,C=GB')")
+    override fun run() = runBlocking {
+        if (pledgeId == null) {
+            println("Arguments required: --pledge-id.")
+        } else if (transferCategory == null) {
+            println("Arguments required: --transfer-category.")
+        } else if (importNetworkId == null) {
+            println("Arguments required: --import-network-id.")
+        } else if (exportRelayAddress == null) {
+            println("Arguments required: --export-relay-address.")
+        } else if (param == null) {
+            println("Arguments required: --param.")
+        } else {
+            val rpc = NodeRPCConnection(
+                host = config["CORDA_HOST"]!!,
+                username = "clientUser1",
+                password = "test",
+                rpcPort = config["CORDA_PORT"]!!.toInt())
+            val thisPartyName = rpc.proxy.startFlow(::GetOurIdentity).returnValue.get()
+            // "thisParty" is set to the token "issuer" in case fungible house token; since we are using the same
+            // SDK function claimPledgeFungibleAsset and Interop application for both the "Simple Asset" and
+            // the "Fungible house token" corDapps, we pass the Identity of the party submitting the claim here.
+            val thisParty = rpc.proxy.wellKnownPartyFromX500Name(CordaX500Name.parse(thisPartyName))!!
+
+            val assetPledgeState = rpc.proxy.startFlow(::GetAssetPledgeStatus, pledgeId!!, importNetworkId!!).returnValue.get() as AssetPledgeState
```

Comment:
Can there be an SDK function for this? Including the safety checks below.

Reply:
Added SDK function, as suggested.